[5021,5101,5102,5184,5185,5186] imeta error handling fixes and tests

SwooshyCueb opened the pull request:
This is a companion PR to irods/irods_client_icommands#161

The following issues are addressed by this PR:
- 5021
- 5101
- 5102
- 5184
- 5185
- 5186

trel commented:
excellent.

please add 5184 in the first line of the first commit message, for consistency.

and one little codacy thing.

alanking commented on scripts/core_tests_list.json:
```diff
@@ -13,6 +13,7 @@
     "test_icp",
     "test_ifsck",
     "test_ils",
+    "test_error_handling",
```
Should this be `test_imeta_error_handling`?

trel replied:
looks like it

korydraughn commented on lib/core/src/rodsPath.cpp:
```diff
@@ -31,6 +31,11 @@ parseRodsPathStr( const char *inPath, rodsEnv *myRodsEnv, char *outPath ) {
     int status;
     rodsPath_t rodsPath;
 
+    if ( strlen(inPath) >= MAX_PATH_ALLOWED - 1) {
```
`inPath` needs to be checked for null before passing to `strlen`.

trel commented:
seems good - have we seen the tests pass?   in CI?  at the bench?

might be time to break out the jenkins repository...

SwooshyCueb commented:
Everything passes in Jenkins except `test_resource_types.Test_Resource_Replication_With_Retry`, which I'm told has unrelated issues.

Rebased against master and added the #s
